### PR TITLE
add Clang `opt` binary to RPATH wrapper dir

### PR DESCRIPTION
**Pull request by jfgrimm**
addresses https://github.com/easybuilders/easybuild-easyblocks/issues/3075

(created using `eb --new-pr`)

### CODE REVIEW

**Comment by jfgrimm** [comment]
Test report by @jfgrimm
#### Overview of tested easyconfigs (in order)
 * **FAIL (unhandled exception: expected str, bytes or os.PathLike object, not tuple)Traceback (most recent call last):
  File "/opt/apps/eb/software/EasyBuild/4.9.0/lib/python3.6/site-packages/easybuild/main.py", line 135, in build_and_install_software
    (ec_res['success'], app_log, err) = build_and_install_one(ec, init_env)
  File "/opt/apps/eb/software/EasyBuild/4.9.0/lib/python3.6/site-packages/easybuild/framework/easyblock.py", line 4262, in build_and_install_one
    result = app.run_all_steps(run_test_cases=run_test_cases)
  File "/opt/apps/eb/software/EasyBuild/4.9.0/lib/python3.6/site-packages/easybuild/framework/easyblock.py", line 4141, in run_all_steps
    self.run_step(step_name, step_methods)
  File "/opt/apps/eb/software/EasyBuild/4.9.0/lib/python3.6/site-packages/easybuild/framework/easyblock.py", line 3976, in run_step
    step_method(self)()
  File "/tmp/eb-q16x5a_3/included-easyblocks-9q8q70f8/easybuild/easyblocks/clang.py", line 754, in sanity_check_step
    super(EB_Clang, self).sanity_check_step(custom_paths=custom_paths, custom_commands=custom_commands)
  File "/opt/apps/eb/software/EasyBuild/4.9.0/lib/python3.6/site-packages/easybuild/framework/easyblock.py", line 3074, in sanity_check_step
    self._sanity_check_step(*args, **kwargs)
  File "/opt/apps/eb/software/EasyBuild/4.9.0/lib/python3.6/site-packages/easybuild/framework/easyblock.py", line 3610, in _sanity_check_step
    self.fake_mod_data = self.sanity_check_load_module(extension=extension, extra_modules=extra_modules)
  File "/opt/apps/eb/software/EasyBuild/4.9.0/lib/python3.6/site-packages/easybuild/framework/easyblock.py", line 3537, in sanity_check_load_module
    self.fake_mod_data = self.load_fake_module(purge=True, extra_modules=extra_modules, verbose=True)
  File "/opt/apps/eb/software/EasyBuild/4.9.0/lib/python3.6/site-packages/easybuild/framework/easyblock.py", line 1700, in load_fake_module
    fake_mod_path = self.make_module_step(fake=True)
  File "/opt/apps/eb/software/EasyBuild/4.9.0/lib/python3.6/site-packages/easybuild/framework/easyblock.py", line 3748, in make_module_step
    txt += self.make_module_req()
  File "/opt/apps/eb/software/EasyBuild/4.9.0/lib/python3.6/site-packages/easybuild/framework/easyblock.py", line 1582, in make_module_req
    paths = sum((glob.glob(path) if path else [path] for path in reqs), [])  # sum flattens to list
  File "/opt/apps/eb/software/EasyBuild/4.9.0/lib/python3.6/site-packages/easybuild/framework/easyblock.py", line 1582, in <genexpr>
    paths = sum((glob.glob(path) if path else [path] for path in reqs), [])  # sum flattens to list
  File "/usr/lib64/python3.6/glob.py", line 20, in glob
    return list(iglob(pathname, recursive=recursive))
  File "/usr/lib64/python3.6/glob.py", line 40, in _iglob
    dirname, basename = os.path.split(pathname)
  File "/usr/lib64/python3.6/posixpath.py", line 107, in split
    p = os.fspath(p)
TypeError: expected str, bytes or os.PathLike object, not tuple
** _Clang-13.0.1-GCCcore-11.3.0-CUDA-11.7.0.eb_ 

Build succeeded for 0 out of 1 (1 easyconfigs in total)
node033.viking2.yor.alces.network - Linux Rocky Linux 8.8, x86_64, AMD EPYC 7643 48-Core Processor, Python 3.6.8
See https://gist.github.com/jfgrimm/4caa1d14c305c64f16ae07deff30256a for a full test report.

File: easybuild/easyblocks/c/clang.py [hunk 1]
```diff
@@ -652,7 +657,7 @@ def sanity_check_step(self):
                 print_warning("Could not find runtime library directory")
                 self.runtime_lib_path = "lib"
         else:
-            self.runtime_lib_path = "lib"
+            self.runtime_lib_path = "lib",
```
Comment (jfgrimm) [inline easybuild/easyblocks/c/clang.py:660]:
```suggestion
            self.runtime_lib_path = "lib"
```

**Comment by jfgrimm** [comment]
Test report by @jfgrimm
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _Clang-13.0.1-GCCcore-11.3.0-CUDA-11.7.0.eb_ 

Build succeeded for 1 out of 1 (1 easyconfigs in total)
node033.viking2.yor.alces.network - Linux Rocky Linux 8.8, x86_64, AMD EPYC 7643 48-Core Processor, Python 3.6.8
See https://gist.github.com/jfgrimm/85b6a44db0e6fd355cc01c16eb74ca2e for a full test report.

**Comment by jfgrimm** [comment]
Test report by @jfgrimm
#### Overview of tested easyconfigs (in order)
 * **FAIL (build issue)** _Clang-10.0.0-GCCcore-8.3.0.eb_ (partial log available at https://gist.github.com/jfgrimm/11e5e9caafafe6ef09d830c794d45941)
 * **SUCCESS** _Clang-11.0.1-GCCcore-10.2.0.eb_ 
 * **SUCCESS** _Clang-12.0.1-GCCcore-10.3.0.eb_ 
 * **SUCCESS** _Clang-13.0.1-GCCcore-11.2.0.eb_ 
 * **SUCCESS** _Clang-13.0.1-GCCcore-11.3.0.eb_ 
 * **SUCCESS** _Clang-15.0.5-GCCcore-11.3.0.eb_ 
 * **SUCCESS** _Clang-16.0.6-GCCcore-12.3.0.eb_ 
 * **SUCCESS** _Z3-4.8.9-GCCcore-9.3.0.eb_ 
 * **SUCCESS** _Z3-4.12.2-GCCcore-12.2.0.eb_ 
 * **SUCCESS** _Clang-11.0.0-GCCcore-9.3.0.eb_ 
 * **SUCCESS** _Clang-16.0.4-GCCcore-12.2.0.eb_ 

Build succeeded for 10 out of 11 (9 easyconfigs in total)
himem01.viking2.yor.alces.network - Linux Rocky Linux 8.8, x86_64, AMD EPYC 7643 48-Core Processor, Python 3.6.8
See https://gist.github.com/jfgrimm/133993586e3a38ea8fcfd0a4921e3dee for a full test report.

**Review by boegel — APPROVED** [review]
lgtm

**Comment by boegel** [comment]
`GCCcore/8.3.0` toolchain is no longer supported, so not blocking this PR over that one test failure...

**Pull request merged**